presto-main/src/main/java/com/facebook/presto/sql/analyzer/StatementAnalyzer.java

```diff
@@ -1594,9 +1594,17 @@ private Scope computeAndAssignOrderByScopeWithAggregation(OrderBy node, Scope ou
                     .collect(toImmutableList());
 
             // generate placeholder fields
+            Set<Field> seen = new HashSet<>();
             List<Field> orderByAggregationSourceFields = orderByAggregationExpressions.stream()
-                    .map(analysis::getType)
-                    .map(type -> Field.newUnqualified(Optional.empty(), type))
+                    .map(expression -> {
+                        // Expose aggregation keys that are also source fields to ORDER BY
+                        Optional<Field> sourceField = sourceScope.tryResolveField(expression)
+                                .filter(ResolvedField::isLocal)
```

Comment:
I don't think it needs to be local. It is subquery planer responsibility to fail if it cannot support such correlated subqueries.